include/avk/cpp_utils.hpp

```diff
@@ -1161,85 +1161,144 @@ namespace avk
 
 #pragma region swap + dispose and handle help functions
 
-	/**
-	*	This concept captures those objects which have a size() method, like vectors	*	
-	*/
+	/**  This concept captures those objects which have a size() method, like vectors
+	 */
 	template <typename T>
 	concept has_size = requires (T x)
 	{
 		x.size();
 	};
 
-	/**
-	*	This concept captures those objects that are explicitely convertible to boolean
-	*  
-	*	Note: most vulkan resources contain an explicit boolean converter
-	*/
+	/** This concept captures those objects that are explicitely convertible to boolean
+	 *
+	 *	Note: most vulkan resources contain an explicit boolean converter.
+	 */
 	template <class T>
 	concept boolean_convertible = requires(T(&f)()) {
 		static_cast<bool>(f());
 	};
 
-	/**
-	* Swaps aOld and aNew and handles the life time of aOld
-	*
-	* The purpose of the functions in this section swap and handle lifetime
-	* in a more clean structure in the source code.
-	*
-	* Additionally, when possible, lifetime_handler+swap is ignored if the
-	* rhs parameter contains no values.
-	*
-	* Be aware that inital value  of aOld  will be disposed of
-	*
-	* after the call, aOld's memory will contain initial aNew's value
-	*
-	* @param aNew				new Resource which will be moved to old Resource's place after operation
-	* @param aOld				old Resource object will be disposed of
-	* @param aLifeTimeHandler	Life time handler for aOld
-	*
-	*/
+	/** Swaps aOld and aNew and handles the life time of aOld.
+	 *
+	 *  Be aware that inital value of aOld will be disposed of.

```

Comment:
function name was changed and description updated as discussed.
Still, writing a description like this in lawyer speech needs some mental gymnastics. Please let me know if it can be improved in your view.